Automatic linting on commit

4rt3ry opened the pull request:
## Description
Added git hook `pre-commit` (invoked by command `git commit`). This hook runs the command `npx eslint --fix` on all staged files, automatically fixing any fixable warnings (such as indents and semicolons) and re-staging the file for commit. 
> This hook is invoked by [git-commit[1]](https://git-scm.com/docs/git-commit), and can be bypassed with the `--no-verify` option. It takes no parameters, and is invoked before obtaining the proposed commit log message and making a commit. Exiting with a non-zero status from this script causes the git commit command to abort before creating a commit. [Documentation](https://git-scm.com/docs/githooks#_pre_commit).

If you want to update any [ESLint rules](https://eslint.org/docs/latest/rules), update the config file located in /eslint.config.mjs.  Keep in mind that many of the styling rules (such as "indent" and "semi") were deprecated and moved to the [@stylistic/js](https://eslint.style/packages/js) plugin. The stylistic rules documentation is located at the bottom of [@stylistic/js](https://eslint.style/packages/js)

As the git hook documentation states, you can run the command `git commit --no-verify` to completely bypass the `pre-commit` hook (and by doing so, bypass linting/formatting).

## Setup
- Run the following command:
`npm run setup` to set up git hooks 

## Future work 
Set up github actions to lint/format pull requests.


This enables git to check the /git_hooks/ directory for any hooks. Currently, `pre-commit` is the only hook set up. This command assumes your /.git/ directory to be located at the top level of the project, in the same directory as /git_hooks/

tstodter commented:
Can we add the git config command to an npm "setup" script? Or whatever you want. Could even experiment with adding it to a "prestart" script, which would cause it to always be fun when `npm start` is run.

tstodter commented:
Will also want to have the linter run on each PR, but suggest that as a follow up PR.

tstodter commented:
And it'd be great to add the `npm setup` requirement to a "how to run this repo" section of the readme.

4rt3ry commented:
# Updates
## Setup
Run the script `npm run setup` to set up git hooks 
- This replaces the old command `git config core.hooksPath ./git_hooks`.

## Future work 
Set up github actions to lint/format pull requests.

BlckHawker commented:
The link you sent does give a list of what rules are avaible, but I'm unsure what `quotes` means, and the documentation doesn't seem to link anywhere. Probably since it's deprecated. The same applies with the other rules as well. I can't properly test them if I'm unable to understand exactly what they do + their parameters

4rt3ry commented:
> The link you sent does give a list of what rules are avaible, but I'm unsure what `quotes` means, and the documentation doesn't seem to link anywhere. Probably since it's deprecated. The same applies with the other rules as well. I can't properly test them if I'm unable to understand exactly what they do + their parameters

I updated the PR documentation to be a little more clear. 
> Keep in mind that many of the styling rules (such as "indent" and "semi") were deprecated and moved to the [@stylistic/js](https://eslint.style/packages/js) plugin. The stylistic rules documentation is located at the bottom of [@stylistic/js](https://eslint.style/packages/js)

Rules such as "quotes" are located in the [@stylistic/js](https://eslint.style/packages/js) plugin. I can add a comment to the link in the /eslint.config.mjs file as well.

BlckHawker commented:
> > The link you sent does give a list of what rules are avaible, but I'm unsure what `quotes` means, and the documentation doesn't seem to link anywhere. Probably since it's deprecated. The same applies with the other rules as well. I can't properly test them if I'm unable to understand exactly what they do + their parameters
> 
> I updated the PR documentation to be a little more clear. 
> > Keep in mind that many of the styling rules (such as "indent" and "semi") were deprecated and moved to the [@stylistic/js](https://eslint.style/packages/js) plugin. The stylistic rules documentation is located at the bottom of [@stylistic/js](https://eslint.style/packages/js)
> 
> Rules such as "quotes" are located in the [@stylistic/js](https://eslint.style/packages/js) plugin. I can add a comment to the link in the /eslint.config.mjs file as well.

Do you mind also updating the description to note that you added a command in `npm run setup` ? It's a little disorienting to have the update be in a comment

4rt3ry commented:
updated description

tstodter commented on package.json:
```diff
@@ -4,6 +4,11 @@
+    "lint:fix": "npx eslint --fix src",
+    "setup": "git config core.hooksPath ./git_hooks",
+    "prestart": "npm run setup",
+    "start": "node ./src/index.js",
```
🙏

tstodter reviewed (approved):
Solid. We'll want to follow up with a more thorough set of linting rules (likely just rip off a standard set), and a definitely separate PR running fix mode on all files.

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExNG11cjBkdjQycHE1a3h0enZ3eGRkdjZzdXRtcWlrOGJvbGNkc3lpZSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/3o7qiIWcK9ZO6cgA3S/giphy-downsized.gif)
